src/scanner.cc

```diff
@@ -228,108 +227,6 @@ namespace {
     }
   }
 
-  // Possible states for the comment lexer to enter.
-  enum CLexState {
-    CLexState_CONSUME,
-    CLexState_ASTERISK,
-    CLexState_L_PAREN,
-    CLexState_LEFT_COMMENT_DELIMITER,
-    CLexState_RIGHT_COMMENT_DELIMITER,
-    CLexState_LOOKAHEAD,
-    CLexState_LOOKAHEAD_NEWLINE,
-    CLexState_LOOKAHEAD_L_PAREN,
-    CLexState_END_OF_FILE
-  };
-  
-  /**
-   * Scans for block comment text. This scanner function supports nested
-   * block comments, so (* text (* text *) text *) will all be parsed as
-   * a single block comment. Also, multiple block comments separated
-   * only by spaces, tabs, or a single newline will be parsed as a single
-   * block comment for convenience; for example, the following is a
-   * single comment:
-   * 
-   *   (***********************)
-   *   (* text text text text *)
-   *   (* text text text text *)
-   *   (* text text text text *)
-   *   (***********************)
-   * 
-   * Although the following constitutes two separate block comments:
-   * 
-   *   (***********************)
-   *   (* text text text text *)
-   *   (***********************)
-   * 
-   *   (***********************)
-   *   (* text text text text *)
-   *   (***********************)
-   * 
-   * A block comment will also be emitted if EOF is reached.
-   *
-   * @param lexer The tree-sitter lexing control structure.
-   * @return Whether any block comment text was detected.
-   */
-  bool scan_block_comment_text(TSLexer* const lexer) {
-    uint32_t nest_level = 0;
-    CLexState state = CLexState_CONSUME;
-    START_LEXER();
-    eof = !has_next(lexer);
-    switch (state) {
-      case CLexState_CONSUME:
-        if (eof) ADVANCE(CLexState_END_OF_FILE);
-        if ('*' == lookahead) ADVANCE(CLexState_ASTERISK);
-        if ('(' == lookahead) ADVANCE(CLexState_L_PAREN);
-        ADVANCE(CLexState_CONSUME);
-        END_STATE();
-      case CLexState_ASTERISK:
-        if ('*' == lookahead) ADVANCE(CLexState_ASTERISK);
-        if ('(' == lookahead) ADVANCE(CLexState_L_PAREN);
-        if (')' == lookahead) ADVANCE(CLexState_RIGHT_COMMENT_DELIMITER);
-        ADVANCE(CLexState_CONSUME);
-        END_STATE();
-      case CLexState_L_PAREN:
-        if ('*' == lookahead) {ADVANCE(CLexState_LEFT_COMMENT_DELIMITER);}
-        if ('(' == lookahead) ADVANCE(CLexState_L_PAREN);
-        ADVANCE(CLexState_CONSUME);
-        END_STATE();
-      case CLexState_LEFT_COMMENT_DELIMITER:
-        nest_level++;
-        GO_TO_STATE(CLexState_CONSUME);
-        END_STATE();
-      case CLexState_RIGHT_COMMENT_DELIMITER:
-        if (nest_level > 0) {
-          nest_level--;
-          GO_TO_STATE(CLexState_CONSUME);
-        } else {
-          ACCEPT_TOKEN(BLOCK_COMMENT_TEXT);
-          if (iswspace(lookahead)) GO_TO_STATE(CLexState_LOOKAHEAD);
```

Comment:
I think one or more of the macros defined at the top of the file like `GO_TO_STATE` or others were only used in this method; could you check and delete the unused macros if so?

Reply:
`GO_TO_STATE` is used in `scan_extramodular_text`. Others are also used